### PR TITLE
fix: fixes evaluation blowing up when referencing local func args

### LINKED ISSUE #281: Babel compilation error when css prop values refer to function arguments
**Describe the bug**
The babel plugin fails to compile when a css prop contains a reference to a function argument, either destructured or primitive arguments. 

**To Reproduce**

```
import React from 'react';

import '@compiled/core';

export const DestructuredComp = ({ foo }) => {
  return (
    <>
      <span css={{ fontSize: foo }} />
    </>
  );
};
```

**Expected behaviour**
No errors

**Actual behaviour**

```
RangeError: /.../packages/design-system/lozenge/src/Lozenge/Test.tsx: Maximum call stack size exceeded
    at Object.exports.getInterpolation (/.../node_modules/@compiled/babel-plugin/dist/utils/ast.js:318:11)
    at Object.exports.getInterpolation (/.../node_modules/@compiled/babel-plugin/dist/utils/ast.js:323:29)
    at Object.exports.getInterpolation (/.../node_modules/@compiled/babel-plugin/dist/utils/ast.js:323:29)
    at Object.exports.getInterpolation (/.../node_modules/@compiled/babel-plugin/dist/utils/ast.js:323:29)
    at Object.exports.getInterpolation (/.../node_modules/@compiled/babel-plugin/dist/utils/ast.js:323:29)
    at Object.exports.getInterpolation (/.../node_modules/@compiled/babel-plugin/dist/utils/ast.js:323:29)
    at Object.exports.getInterpolation (/.../node_modules/@compiled/babel-plugin/dist/utils/ast.js:323:29)
    at Object.exports.getInterpolation (/.../node_modules/@compiled/babel-plugin/dist/utils/ast.js:323:29)
    at Object.exports.getInterpolation (/.../node_modules/@compiled/babel-plugin/dist/utils/ast.js:323:29)
    at Object.exports.getInterpolation (/.../node_modules/@compiled/babel-plugin/dist/utils/ast.js:323:29) {
  code: 'BABEL_TRANSFORM_ERROR'
}
```

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by madou**
Closes #281

### CODE REVIEW

File: packages/babel-plugin/src/utils/ast.tsx [hunk 1]
```diff
@@ -260,7 +260,9 @@ export const resolveBindingNode = (
   binding: Binding | undefined,
   meta: Metadata
 ): PartialBindingWithMeta | undefined => {
-  if (!binding) {
+  if (!binding || binding.path.isObjectPattern()) {
```
Comment (pgmanutd) [inline packages/babel-plugin/src/utils/ast.tsx:263]: 🤔 Can there be more use cases like this? Just thinking 🤔 If yes, then this **or** condition will go up 😞

Comment (madou) [reply packages/babel-plugin/src/utils/ast.tsx:263]: most likely tbh - but im sure its a finite list

Comment (pgmanutd) [reply packages/babel-plugin/src/utils/ast.tsx:263]: > most likely tbh - but im sure its a finite list
> 
> can we think of any other scenarios that we've missed (when it comes to bindings referencing something in scope)? i can't think of any.

I just pulled the branch. There are 2 things:

1. It's still throwing "Maximum call stack size exceeded" error for this use case 😞
```typescript
import '@compiled/core';

export const DestructuredComp = (props) => {
  const { foo, color } = props;

  return (
    <>
      <span css={{ fontSize: foo, color, backgroundColor: 'blue' }} />
    </>
  );
};
```
2. We are skipping the resolving part right, but what if its coming from a variable outside the function 🤔? May be from a variable declarator or import. Should we also skip in that case 🤔? Something like this:

```typescript
import '@compiled/core';

const { foo } = { foo: 10 };

export const DestructuredComp = ({ color }) => {
  return (
    <>
      <span css={{ fontSize: foo, color, backgroundColor: 'blue' }} />
    </>
  );
};
```

Comment (madou) [reply packages/babel-plugin/src/utils/ast.tsx:263]: awesome insight man, i'll do some testing this morning and fix anything that breaks. we should still be supporting proper de-structuring - perhaps we don't right now?

good catch for the maximum call stack exceed - i'll write a text

File: packages/babel-plugin/src/utils/ast.tsx [hunk 2]
```diff
@@ -372,6 +374,10 @@ export const getInterpolation = (expression: t.Expression, meta: Metadata): t.Ex
   if (t.isIdentifier(expression)) {
     const binding = meta.parentPath.scope.getBinding(expression.name);
     const resolvedBinding = resolveBindingNode(binding, meta);
+    if (binding?.path.node === expression) {
```
Comment (pgmanutd) [inline packages/babel-plugin/src/utils/ast.tsx:377]: hehe 😛. Breaking the circular cycle with reference check. Nice.

**Review by pgmanutd — APPROVED** [review]

File: packages/babel-plugin/src/utils/ast.tsx (hunk 2 above)
Comment (madou) [reply packages/babel-plugin/src/utils/ast.tsx:377]: yeah, elegant!

**Pull request merged**